Source/Core/Core/HW/DSPHLE/DSPHLE.cpp

```diff
@@ -191,26 +191,37 @@ void DSPHLE::DSP_WriteMailBoxLow(bool cpu_mailbox, u16 value)
 // Other DSP functions
 u16 DSPHLE::DSP_WriteControlRegister(u16 value)
 {
-  DSP::UDSPControl temp(value);
```

Comment:
Actually I think I misread it, in each path there's only one change of cr so a temporary won't help much - as to keep the same behavior in case the register is read elsewhere somehow you might need to write back control anyway (as seen below) so there's not much benefit in this case. Shouldn't have bothered you, sorry (though the point is legitimate for the other function).

```
u16 DSPHLE::DSP_ReadControlRegister()
{
  u32 crc = UCodeInterface::GetCRC(m_ucode.get());
  u16 control = m_dsp_control.Hex;
  if (crc == UCODE_ROM || crc == UCODE_INIT_AUDIO_SYSTEM)
    control |= 0x800;
  else
    control &= ~0x800;
  m_dsp_control.hex = control;
  return control;
}
```